src/gui/guiFormSpecMenu.cpp

```diff
@@ -3767,14 +3772,46 @@ void GUIFormSpecMenu::updateSelectedItem()
 			if (item.empty())
 				continue;
 
-			// Grab selected item from the crafting result list
-			m_selected_item = new GUIInventoryList::ItemSpec;
-			m_selected_item->inventoryloc = e->getInventoryloc();
-			m_selected_item->listname = "craftresult";
-			m_selected_item->i = 0;
-			m_selected_item->slotsize = e->getSlotSize();
-			m_selected_amount = item.count;
-			m_selected_dragging = false;
+			GUIInventoryList::ItemSpec s = GUIInventoryList::ItemSpec();
+			s.inventoryloc = e->getInventoryloc();
+			s.listname = "craftresult";
+			s.i = 0;
+			s.slotsize = e->getSlotSize();
+
+			if (m_shift_craft) {
+				// Try to shift-move the crafted item somewhere,
+				// but not back into the crafting grid
+				for (u32 i = 0; i < m_inventory_rings.size(); i++) {
+					const ListRingSpec &lr = m_inventory_rings[i];
```

Comment:
This likely works well in Minetest Game but if another game defines list rings it might be worse than hardcoding "main" as the destination list (which is de-facto mandatory).

Reply:
Do you have any suggestions? Would it be better to look for the next list in the ring after "craft"?

Reply:
I believe any guesswork overall can result in buggy behaviour, so I would explicitly look for the next list after "craftresult" in the list ring. Default to "main" if there is no such rule given by games.

Reply:
But "craftresult" isn't in the list ring, because it shouldn't be part of the formspec, which is where the list rings are defined.

I've adjusted the code and added a comment explaining the reasoning. If there is no list after "craft" then the item is selected as normal.

Reply:
> It's part of the formspec though, where else would it be?

No, the "craft" (the 3x3 grid) and "craftpreview" (what you click to craft) lists are part of the formspec, but the "craftresult" list is not.

The "craftresult" list shouldn't be part of the formspec because you never need to interact with it directly, and in fact there is code preventing items being moved there: https://github.com/minetest/minetest/blob/180ec92ef982d9fb5c6bdc789f381335f77823c1/src/network/serverpackethandler.cpp#L682-L690